cadc-storage-adapter-ad/src/main/java/org/opencadc/inventory/storage/ad/AdStorageQuery.java

```diff
@@ -134,24 +134,23 @@ public AdStorageMetadataRowMapper() { }
         public StorageMetadata mapRow(List<Object> row) {
             Iterator i = row.iterator();
 
-            String archive = (String) i.next();
-            String fname = (String) i.next();
             URI uri = (URI) i.next();
             if (uri == null) {
-                log.warn(AdStorageMetadataRowMapper.class.getSimpleName() + ".SKIP loc=" + archive + "/" + fname + " reason=null-uri");
+                log.warn(AdStorageMetadataRowMapper.class.getSimpleName() + ".SKIP reason=null-uri");
                 return null;
             }
             
+            // trust uri, <scheme>:<archive>/<fname>
+            String archiveFname = uri.toString().split(":")[1];
+            final String archive = archiveFname.split("/")[0];
+            
             // chose best storageID
-            URI sid = URI.create("ad:" + archive + "/" + fname);
-            if ("mast".equals(uri.getScheme())) {
-                sid = uri;
-            }
+            URI sid = uri;
```

Comment:
This `sid` variable is now superfluous; in fact, you could now declare storageID way up at line 137. The easy way to do that is to remove lines 147-149, then rename `uri` to `storageID` on 137 (with IDE refactoring tool, it will rename all uses as well).

Reply:
Done.